### ISSUE TITLE
Feature Request: Automatic Keep-Alive

### ISSUE THREAD

**Issue by nikeee** [nikeee/TeamSpeak3QueryAPI#65]
See #64 for the need of this.

# Option 1:
Use `System.Timers.Timer`
Pros:
- Simple to implement
- Can be implemented external from the TeamSpeak client without breaking changes
Cons:
- We currently target Net Standard 1.3 - there might not be a Timer available, so we have to bump the Net Standard version.

Note:
- Which Timer to use? See: https://jonskeet.uk/csharp/threads/timers.html
- We don't want to force the library user to use a specific one

# Option 2:
Timeout in the response queue.
Pros:
- KeepAlive will only be sent if there was no 
Cons:
- Harder to implement
Note:
- Must be implemented in the QueryClient itself; if done correctly, would be transparent to the user

# Option N
[your suggestion here (comment on this issue)]

# General implementation notes:
- Some library users are have already implemented their own Keep-Alive logic.
- Design decision: Keep the query client / rich client dumb?

**Comment by Arefu** [comment]
[Looks like I was wrong](https://docs.microsoft.com/en-us/dotnet/api/system.timers.timer?view=netcore-3.1#moniker-applies-to), Timers were introduced in .NET Standard 2.0.

I'll take a look at implementing it internally, I agree it should mostly be transparent, but there should be ways to disable it in case the user already has their own solution in place.

**Comment by Des1re7** [comment]
I have implemented this as a simple task:
```
private async Task KeepAlive(CancellationToken cancellationToken, TimeSpan delay)
{
	while (!cancellationToken.IsCancellationRequested)
	{
		await TeamSpeak.WhoAmI();
		await Task.Delay(delay, cancellationToken);
	}
}
```

**Comment by nikeee** [comment]
Solved via #67
